Add gen-dogecoin-conf.sh from bitcoin core github

GaloisField2718 opened the pull request:
I updated the bitcoin script [gen-bitcoin-conf.sh](https://github.com/bitcoin/bitcoin/tree/master/contrib/devtools/gen-bitcoin-conf.sh) to generate a template dogecoin conf file. 

The file is at: `./contrib/devtools/gen-dogecoin-conf.sh` and generate the conf file into `./share`. As there is not `./share/examples` directory I just let it in `./share`.

patricklodder commented on share/dogecoin.conf:
```diff
@@ -0,0 +1,463 @@
+# datadir/backups)
+#backupdir=<dir>
+
+# Specify configuration file (default: dogecoin.conf)
+#conf=<file>
```
minor: file would be referencing itself? May want to create an exclusion for this

GaloisField2718 replied:
It's the case for bitcoin.conf in fact. Indeed sounds weird but maybe if you have multiple configs you can avoid to specify the conf with putting one in default path and redirect from conf to the other. 
It can simplify the command maybe.

patricklodder replied:
Fair enough. I'll test whether that works separate from this PR - this is non-blocking for now.

GaloisField2718 replied:
It's also on Bitcoin because to remove it we would need to create a specific regex or remove this line at the end of the execution. 
Don't know what can be better but it's not done on gen-bitcoin-conf.sh. 
Do you think that it's something I should try to do?

patricklodder replied:
I'd let it be for now. If this doesn't work we can also decide to change the behavior - I understand the use-case you describe for it. 

Consider this no problem for acceptance from my end, but let's keep it visible as something that we'd ideally investigate at some point before we release 1.15.0

GaloisField2718 commented on contrib/devtools/gen-dogecoin-conf.sh:
```diff
@@ -0,0 +1,83 @@
+# Options for testnet
+[test]
+
+# Options for signet
+[signet]
+
```
Ok it's weird because it should be generated by the help from dogecoind. I'll check this to update it thanks.

GaloisField2718 replied:
> There is no signet in Dogecoin

I updated it. Indeed it's the footer generated by the script. I didn't see it previously.

patricklodder reviewed (changes requested):
- script is good
- example file needs an update to reflect the latest changes to the script
- could use a squash after that - feel free to squash it all into a single commit

GaloisField2718 commented:
> * script is good
> * example file needs an update to reflect the latest changes to the script
> * could use a squash after that - feel free to squash it all into a single commit

Thanks! I updated the example file and squashed everything into one commit. 

Thanks for your comments and help!

GaloisField2718 commented:
I think I made a small mistake with squashing because your last commit is also appearing in this PR. 
I don't know how to remove it and if it's a problem.

patricklodder commented:
I'll fix it for you - it happens because you're working off `master` whereas we're pulling this against `1.15.0-dev`.

patricklodder reviewed (approved):
ACK - tested the script to work, reviewed the output

Needs a 2nd check from another maintainer because I force-pushed it to resolve a conflict in the commit tree

GaloisField2718 commented:
Ok. Thank you for your intervention

chromatic commented:
I'm generally okay with this, but I see one potential change from running it locally myself:

```
$ git diff
diff --git a/share/dogecoin.conf b/share/dogecoin.conf
index 1f442e702e..dea6953226 100644
--- a/share/dogecoin.conf
+++ b/share/dogecoin.conf
@@ -64,7 +64,7 @@
 # (default: 100)
 #blockreconstructionextratxn=<n>
 
-# Set the number of script verification threads (-10 to 16, 0 = auto, <0 =
+# Set the number of script verification threads (-4 to 16, 0 = auto, <0 =
 # leave that many cores free, default: 0)
 #par=<n>
 
@@ -343,7 +343,7 @@
 # Send trace/debug info to console instead of debug.log file
 #printtoconsole=1
 
-# Shrink debug.log file on client startup (default: 1 when no -debug)
+# Shrink debug.log file on startup/SIGHUP (default: 1 when no -debug)
 #shrinkdebugfile=1
```

GaloisField2718 commented:
> I'm generally okay with this, but I see one potential change from running it locally myself:
> 
> 
> 
> ```
> 
> $ git diff
> 
> diff --git a/share/dogecoin.conf b/share/dogecoin.conf
> 
> index 1f442e702e..dea6953226 100644
> 
> --- a/share/dogecoin.conf
> 
> +++ b/share/dogecoin.conf
> 
> @@ -64,7 +64,7 @@
> 
>  # (default: 100)
> 
>  #blockreconstructionextratxn=<n>
> 
>  
> 
> -# Set the number of script verification threads (-10 to 16, 0 = auto, <0 =
> 
> +# Set the number of script verification threads (-4 to 16, 0 = auto, <0 =
> 
>  # leave that many cores free, default: 0)
> 
>  #par=<n>
> 
>  
> 
> @@ -343,7 +343,7 @@
> 
>  # Send trace/debug info to console instead of debug.log file
> 
>  #printtoconsole=1
> 
>  
> 
> -# Shrink debug.log file on client startup (default: 1 when no -debug)
> 
> +# Shrink debug.log file on startup/SIGHUP (default: 1 when no -debug)
> 
>  #shrinkdebugfile=1
> 
> ```

Hello, 

It means that you don't have the same configuration file? 

Is it related to dogecoin-cli version?

GaloisField2718 commented:
I can put the more complete configuration file you generated @chromatic 
Otherwise, for each build it will generate the according conf file for the user.

chromatic commented:
I ran `make clean` then reconfigured and built again, and now I only have the script verification threads difference. I think that's probably okay.

GaloisField2718 commented:
Ah okay. I maybe don't setup this. I don't know if it's by default, I don't build gui in general for example.

chromatic commented:
Yeah, the differing value in that message comes from a function called `GetNumCores()`, so if I build this on a laptop versus a server, I'll see different output. I think that's fine.

The other difference came from me building from a dirty working tree for another PR, so that was unrelated noise.

GaloisField2718 commented:
Ah yes okay I built only from server. But if anyone runs it from laptop it will generate the option

chromatic reviewed (approved):
ACK, all tests pass on x86-64, code looks fine, and generated file differences make sense based on the output of the system used to generate the example file.
